src/server/run_sched.c

```diff
@@ -253,94 +202,69 @@ find_sched_from_sock(int sock)
- * @param[in]	jobid	- the jobid to send if 'cmd' is SCH_SCHEDULE_AJOB
+ * @return	void
  */
+void
+set_sched_state(pbs_sched *psched, char *state)
+{
+	set_attr_generic(&(psched->sch_attr[SCHED_ATR_sched_state]), &sched_attr_def[SCHED_ATR_sched_state], state, NULL, SET);
+	server.sv_attr[SVR_ATR_State].at_flags |= ATR_VFLAG_MODCACHE;
+}
 
+/**
+ * @brief
+ * 	Receives end of cycle notification from the corresponding Scheduler
+ *
+ * @param[in] sock - socket to read
+ *
+ * @return int
+ * @retval 0  - on success
+ * @retval !0 - on error
+ */
 int
-contact_sched(int cmd, char *jobid, pbs_net_t pbs_scheduler_addr, unsigned int pbs_scheduler_port)
+recv_sched_cycle_end(int sock)
 {
-	int sock;
-	conn_t *conn;
-	struct sigaction act, oact;
-
-	if ((cmd == SCH_SCHEDULE_AJOB) && (jobid == NULL))
-		return -1;	/* need a jobid */
-
-	/* connect to the Scheduler */
-	/* put a timeout alarm around the connect */
-
-	sigemptyset(&act.sa_mask);
-	act.sa_flags = 0;
-	act.sa_handler = catchalrm;
-	if (sigaction(SIGALRM, &act, &oact) == -1)
-		return (PBS_NET_RC_RETRY);
-	alarm(SCHEDULER_ALARM_TIME);
-
-	/* This function does a timeout wait on the non-blocking socket */
-	sock = client_to_svr(pbs_scheduler_addr, pbs_scheduler_port, B_RESERVED); /* scheduler connection still uses resv-ports */
-	if (pbs_errno == PBSE_NOLOOPBACKIF)
-		log_err(PBSE_NOLOOPBACKIF, "client_to_svr" , msg_noloopbackif);
-
-	alarm(0);
-
-	(void)sigaction(SIGALRM, &oact, NULL);	/* reset handler for SIGALRM */
-	if (sock < 0) {
-		log_err(errno, __func__, msg_sched_nocall);
-		return (-1);
-	}
-	conn = add_conn_priority(sock, FromClientDIS, pbs_scheduler_addr,
-		pbs_scheduler_port, NULL, process_request, PRIORITY_CONNECTION);
-	if (!conn) {
-		log_err(errno, __func__, "could not find sock in connection table");
-		return (-1);
-	}
-	conn->cn_authen |= PBS_NET_CONN_TO_SCHED | PBS_NET_CONN_FROM_PRIVIL | PBS_NET_CONN_AUTHENTICATED;
-	strcpy(conn->cn_username, PBS_SCHED_DAEMON_NAME);
-	conn->cn_credid = strdup(PBS_SCHED_DAEMON_NAME);
-	if (conn->cn_credid == NULL) {
-		log_err(errno, __func__, "Out of memory!");
-		return (-1);
-	}
-	conn->cn_auth_config = make_auth_config(AUTH_RESVPORT_NAME, "",
-							pbs_conf.pbs_exec_path,
-							pbs_conf.pbs_home_path,
-							(void *)log_event);
-	if (conn->cn_auth_config == NULL) {
-		log_err(errno, __func__, "Out of memory!");
-		return -1;
-	}
-	DIS_tcp_funcs();
-	transport_chan_set_ctx_status(sock, AUTH_STATUS_CTX_READY, FOR_AUTH);
-	net_add_close_func(sock, scheduler_close);
+	pbs_sched *psched;
 
-	if (set_nodelay(sock) == -1) {
-		snprintf(log_buffer, sizeof(log_buffer), "cannot set nodelay on connection %d (errno=%d)\n", sock, errno);
-		log_err(-1, __func__, log_buffer);
-		return (-1);
-	}
+	for (psched = (pbs_sched*) GET_NEXT(svr_allscheds); psched; psched = (pbs_sched*) GET_NEXT(psched->sc_link)) {
+		if (psched->sc_secondary_conn == sock) {
+			int rc;
+
+			DIS_tcp_funcs();
+			psched->sc_cycle_started = disrsi(sock, &rc);
```

Comment:
What is the purpose of this?  Why do we need to get this information from the scheduler?

Reply:
this is cycle end notification, as we are not closing connection to notify server like earlier. Please see send_cycle_end() in pbs_sched.c

Reply:
I guess my confusion is why we're reading something into this variable.  Is it possible that we might read true or will we be always be setting it to false?

Reply:
I will change this to make more readable, actually since current cycle_end_marker is 0 and here we have to make 'sc_cycle_started' 0 as we got end cycle notification, i just combined those to in one like, ideally it should have been:
```c
disrsi(sock, &rc); /* read end cycle marker and ignore as we don't need its value */
psched->sc_cycle_started = 0;
// check for rc, and make sched state based on rc - aka SC_DOWN or SC_IDLE
```